File: JAVA/2024-01-03/181851.java
//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
import java.util.*;

public class Main {
    public static void main(String[] args) {
    }
}
public int solution(int[] rank, boolean[] attendance) {
    int answer = 0;
    Map<Integer, Integer> ranker = new HashMap<>();
    for (int i = 0; i < rank.length; i++)
    {
        if (attendance[i])
        {
            ranker.put(rank[i], i);
        }
    }
    List<Integer> ranks = new ArrayList<>(ranker.keySet());
    answer = (10000 * ranker.get(ranks.get(0)) + (100 * ranker.get(ranks.get(1))) + ranker.get(ranks.get(2)));

    return answer;
}
}

File: C++/2025-01-03/181851.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int solution(vector<int> rank, vector<bool> attendance) 
{
    int answer = 0;
    map<int, int> rank_index;

    for (int i = 0; i < rank.size(); i++)
    {
        if (attendance[i])
            rank_index.insert(make_pair(rank[i], i));
    }

    auto p = rank_index.begin();
    answer += 10000 * p++->second;
    answer += 100 * p++->second;
    answer += p->second;

    return answer;
}
